app/api/routes/image_queries.py

```diff
@@ -108,26 +148,54 @@ async def post_image_query(
         },
     )
 
+    img_numpy = np.asarray(image)  # [H, W, C=3], dtype: uint8, RGB format
+
+    edge_inference_manager = app_state.edge_inference_manager
+    motion_detection_manager = app_state.motion_detection_manager
+    require_human_review = human_review == "ALWAYS"
+
     # TODO: instead of just forwarding want_async calls to the cloud, facilitate partial
     #       processing of the async request on the edge before escalating to the cloud.
     _want_async = want_async is not None and want_async.lower() == "true"
     if _want_async:
-        return safe_call_api(
-            gl.submit_image_query,
-            detector=detector_id,
-            image=image,
-            wait=0,
+        edge_only = os.environ.get("EDGE_ONLY", "") == "ENABLED"
+        if not edge_only:
+            # Escalate to cloud api as normal
+            return safe_call_api(
+                gl.submit_image_query,
+                detector=detector_id,
+                image=image,
+                wait=0,
+                patience_time=patience_time,
+                confidence_threshold=confidence_threshold,
+                human_review=human_review,
+                want_async=True,
+            )
+
+        # Get the result of the edge model and return it
+        # NOTE: because this is temporary code, this skips handling motion detection
+
+        if not edge_inference_manager.inference_is_available(detector_id=detector_id):
+            raise ValueError("EDGE_ONLY is set to ENABLED, but edge inference is not available.")
+
+        logger.debug(
+            "EDGE_ONLY is set to ENABLED. This async request will not be escalated to the cloud and the "
+            "edge model's answer will be returned regardless of confidence. Motion detection will be ignored."
+        )
+
+        detector_metadata: Detector = get_detector_metadata(detector_id=detector_id, gl=gl)
+        results = edge_inference_manager.run_inference(detector_id=detector_id, img_numpy=img_numpy)
```

Comment:
We shouldnt have to duplicate the call to run_inference here - I think this logic can be simplified.

As implemented, you're actually cutting motion detection out flow for this path.

The main difference with this new flag is that if it is enabled:
1. we bypass the want_async if statement
2. we return a value whether or not the edge inference model is confident enough.
3. if edge inference is not available, we throw an error

Reply:
The reason I did it this way is because this is just temporary code for the demo, so I tried to avoid making modifications to the existing flow as much as possible. Leo also mentioned it might be simplest/most flexible to make the flag work only for `ask_async`, rather than all sdk calls. 

Since I'm not familiar with this code, I was worried there might be additional things to handle if the flag is incorporated more into the existing logic flow. I looked at the motion detection code and it seems like there's a chance it'll make a cloud API call - although looking at it again, it might not wait for a response, so maybe a delay wouldn't happen?

If we need motion detection for the demo and are okay with EDGE_ONLY being relevant to all api calls, I think it can be reworked as you suggested.

Reply:
I think my suggestion has a couple of strong points:
1. Fewer LOC changed for a demo (temporary code is sort of a myth IMO)
2. Less duplicated code, more consistent with existing behavior (its not really about the motion detection... until it is for some future demo or use case, then we're all like "why doesnt this work for motion detection anymore?")
3. Most importantly: makes it very obvious what the flag is actually doing differently - so contributors don't have to go line-by-line down the divergent code paths to figure it out for themselves.

If motion detection is making a call to the cloud, that is a separate issue we should address, rather than excluding motion detection altogether.

Reply:
That makes sense to me, I'm down to rework it along those lines.

Reply:
I think I'm mostly with Tyler on this one. Skipping motion detection seems like a bug. I don't quite yet visualize what the full change is, so I'm not yet convinced that it will be fewer lines of code, but it certainly will be more correct.

Maybe a tangent, but I recall we were once in conversation with a potential partner who wanted to be able to use Groundlight to train models, but then they wanted to be able to cut out our cloud api costs once the model was trained. Things didn't work out at the time, but if someone like that came back to us I could see a business path where they keep buying hubs so long as the hubs can go fully offline at some point. They would be acutely interested in this behavior and then like Tyler said, the illusion of temporary code bursts